pep-9999.rst

```diff
@@ -0,0 +1,210 @@
+``Py_DEPRECATED`` macro.
+
+* Py_UNICODE_WSTR_LENGTH
+* PyUnicode_GET_SIZE
+* PyUnicode_GET_DATA_SIZE
+* PyUnicode_AS_UNICODE
+* PyUnicode_AS_DATA
+* PyUnicode_AsUnicode
+* _PyUnicode_AsUnicode
+* PyUnicode_AsUnicodeAndSize
+* PyUnicode_FromUnicode
+
+
+Python 3.10
+-----------
+
+Behavior changed:
+
+* ``PyUnicode_FromUnicode(NULL, size)`` and
+  ``PyUnicode_FromStringAndSize(NULL, size))`` will raise
+  DeprecationWarning when ``size > 0``.
+
+* PyArg_ParseTuple and PyArg_ParseTupleAndKeywords will raise
+  DeprecationWarning when 'u', 'u#', 'Z', and 'Z#' format are used.
+
+
+Python 3.12
+-----------
+
+Following members will be removed from Unicode strucutre:
+
+* ``wstr``
+* ``wstr_length``
+* ``state.compact``
+* ``state.ready``
+
+Following macros and functions will be removed:
+
+* Py_UNICODE_WSTR_LENGTH
+* PyUnicode_GET_SIZE
+* PyUnicode_GET_DATA_SIZE
+* PyUnicode_AS_UNICODE
+* PyUnicode_AS_DATA
+* PyUnicode_AsUnicode
+* _PyUnicode_AsUnicode
+* PyUnicode_AsUnicodeAndSize
+* PyUnicode_FromUnicode
+
+Behavior changed:
+
+* ``PyUnicode_FromStringAndSize(NULL, size))`` will raise
+  RuntimeError when ``size > 0``.
+
+* PyArg_ParseTuple and PyArg_ParseTupleAndKeywords will raise
+  RuntimeError when 'u', 'u#', 'Z', and 'Z#' formats are used.
+
+Following macros, enum members will be marked deprecated.
+
+* PyUnicode_WCHAR_KIND
+* PyUnicode_READY -- returns 0 always
+* PyUnicode_IS_READY -- returns 1 always
+* PyUnicode_IS_COMPACT -- returns 1 always
+
+
+Python 3.14
+-----------
+
+Following macros, enum members, and functions will be removed.
+
+* PyUnicode_WCHAR_KIND
+* PyUnicode_READY
+* PyUnicode_IS_READY
+* PyUnicode_IS_COMPACT
+
+Behavior changed:
+
+* PyArg_ParseTuple and PyArg_ParseTupleAndKeywords will raise
+  SystemError instead of RuntimeError, like regular unsupported
+  format.
+
+
+Alternative Ideas
+=================
+
+Advanced Schedule 1
+-------------------
+
+Accelerate all changes by 1 version.
+
+If many third packages are broken by this change, we will revert the
+change and back to regular schedule.
+
+Pros: There is a chance to remove ``wstr`` in Python 3.11. Even if
+we need to revert it, third party maintainers can have more time to
+prepare the removal and we can get feedback from the community early.
+
+Cons: Adding warnings in beta period will make some confusion and
+risk. Note that 'u', 'u#', 'Z', and 'Z#' are used in CPython yet.
+
+
+Advanced Schedule 2
+-------------------
+
+Accelerate only removing legacy Unicode support.
+
+* ``PyUnicode_FromUnicode(NULL, size)`` and
+  ``PyUnicode_FromStringAndSize(NULL, size))`` will raise
+  DeprecationWarning when ``size > 0`` since Python 3.9,
+  and raise RuntimeError since Python 3.11.
+
+* ``state.compact`` and ``state.ready`` will be removed in Python
+  3.11.
+
+This plan is safer than Advanced Schedule 1, because the legacy
+Unicode object is not used in CPython core and stdlib.
+
+
+Use hashtable to store wstr
+---------------------------
```

Comment:
I proposed that when I misunderstood the PEP. You may remove this section, I'm not sure that it's relevant :-)